src/Expander/Primitives.hs

```diff
@@ -486,7 +486,7 @@ log t dest stx = do
 whichProblem :: ExprPrim
 whichProblem t dest stx = do
   unify dest (tMacro (primitiveDatatype "Problem" [])) t
-  Stx _ _ (_ :: Syntax) <- mustHaveEntries stx
+  Stx _ _ (Identity _) <- mustHaveEntries stx
```

Comment:
I have changed the `mustHaveEntries` API so that a singleton tuple is now represented via an `Identity` wrapper and not just by the element itself. One nice consequence of that is that a type annotation is no longer needed in order to drive the type class resolution algorithm.

Reply:
Sounds reasonable!